<launch>
  <arg name="ai_name" default="/blue" />
  <arg name="color" default="blue" />
  <arg name="side" default="left" />

  <group ns="$(arg ai_name)" >
    <param name="ai_name" type="string" value="$(arg ai_name)" />
    <param name="friend_color" type="string" value="$(arg color)" />
    <param name="team_side" type="string" value="$(arg side)" />

    <rosparam command="load" file="$(find ai_core)/param/our_team.yaml" />

    <!-- <include file="$(find ai_core)/launch/simulator.launch" /> -->
    <include file="$(find ai_core)/launch/real.launch" />

    <!-- Run Refbox -->
    <node name="refbox" pkg="ssl_refbox_wrapper" type="ssl_refbox_wrapper_node.py" respawn="true" output="screen">
      <param name="multicast_addr" value="224.5.23.1" />
      <param name="multicast_port" value="10003" />
    </node>

    <!-- Run the map server -->
    <node pkg="tf" type="static_transform_publisher" name="enemy_goal_left" args="6.0 0.5 0 0 0 0 1 map enemy_goal_left 100" />
    <node pkg="tf" type="static_transform_publisher" name="enemy_goal_center" args="6.0 0 0 0 0 0 1 map enemy_goal_center 100" />
    <node pkg="tf" type="static_transform_publisher" name="enemy_goal_right" args="6.0 -0.5 0 0 0 0 1 map enemy_goal_right 100" />
    <node pkg="tf" type="static_transform_publisher" name="friend_goal_left" args="-6.0 0.5 0 0 0 0 1 map friend_goal_left 100" />
    <node pkg="tf" type="static_transform_publisher" name="friend_goal_center" args="-6.0 0 0 0 0 0 1 map friend_goal_center 100" />
    <node pkg="tf" type="static_transform_publisher" name="friend_goal_right" args="-6.0 -0.5 0 0 0 0 1 map friend_goal_right 100" />

    <node name="ball_observer" pkg="world_observer" type="world_observer" respawn="true">
      <param name="observe_target" type="string" value="Ball" />
    </node>

    <node name="world_model" pkg="aisaac" type="world_model.py" output="screen" respawn="true"/>

    <node name="launch_manager" pkg="launch_manager" type="launch_manager.py" respawn="true"/>

    <!-- Visualizer -->
    <node name="sai_visualizer" pkg="sai_visualizer" type="sai_visualizer" respawn="true"/>

    <node name="calculation" pkg="aisaac" type="calculation.py" output="screen" respawn="true"/>
  </group>
</launch>
